src/ex6/client/src/reducers/items-entities-reducer.js

```diff
@@ -0,0 +1,41 @@
+import ACTIONS from "../actions/constants/index";
+
+const initialState = { todos: [], loading: true, oldTodos: [] };
+
+const itemsEntitiesReducer = (state = initialState, action) => {
+  switch (action.type) {
+    case "FETCH_ALL_TODOS_SUCCESS":
+      return { ...state, todos: action.todos, loading: false };
+    case "FETCH_ALL_TODOS_FAILURE":
+      return { ...state, todos: null, loading: false };
+    case "ADD_TODO":
+      const { newItem } = action;
+      const newTodos = [...state.todos, newItem.newTodo];
+      return { ...state, todos: newTodos };
+    case "REMOVE_TODO":
+      const { deletedTodo } = action.id;
+      const filteredTodos = [...state.todos].filter((todo) => {
+        if (todo.id !== deletedTodo) {
+          return todo;
+        }
+      });
+      return { ...state, todos: filteredTodos, oldTodos: [...state.todos] };
+    case "REMOVE_ALL_TODOS":
+      return { ...state, todos: [] };
+    case "CHANGE_TODO_STATUS":
+      const updatedState = [...state.todos].map((todo) => {
```

Comment:
```suggestion
      const updatedState = state.todos.map((todo) => {
```

when you use array function like map, filter etc. it creats a new ref .no need to create another ref from the state, can just go over the original one